code/game/gamemodes/revolution/revsquad.dm

```diff
@@ -0,0 +1,309 @@
+	else
+		to_chat(mob, "The [T] in your [where] will help you to persuade the crew to join your cause.")
+		if(istype(T, /obj/item/device/flash/revsquad))
+			var/obj/item/device/flash/revsquad/FR = T
+			to_chat(mob, "<span class = 'warning'>Your [FR] has [FR.limited_conversions] uses for conversions, and not all of your comrades have one like it. Use it wisely.</span>")
+		mob.update_icons()
+		stat_collection.revsquad.revsquad_items += T.name
+		return 1
+
+/datum/game_mode/revsquad/proc/check_rev_victory()
+	for(var/datum/mind/rev_mind in head_revolutionaries)
+		for(var/datum/objective/objective in rev_mind.objectives)
+			if(!(objective.check_completion()))
+				return 0
+
+		return 1
+
+/datum/game_mode/revsquad/proc/check_heads_victory()
+	for(var/datum/mind/rev_mind in head_revolutionaries)
+		var/turf/T = get_turf(rev_mind.current)
+		if((rev_mind) && (rev_mind.current) && (rev_mind.current.isDead()) && T && (T.z == map.zMainStation))
+			if(ishuman(rev_mind.current))
+				return 0
+	return 1
+
+
+/datum/game_mode/revsquad/check_win()
+	if(check_rev_victory())
+		finished = REVSQUAD_VICTORY_REVS
+	else if(check_heads_victory())
+		finished = REVSQUAD_VICTORY_HEADS
+	return
+
+///////////////////////////////
+//Checks if the round is over//
+///////////////////////////////
+/datum/game_mode/revsquad/check_finished()
+	if(config.continous_rounds)
+		if(finished != 0)
+			if(emergency_shuttle)
+				emergency_shuttle.always_fake_recall = 0
+		return ..()
+	if(finished != 0)
+		return 1
+	else
+		return 0
+
+/datum/game_mode/revsquad/declare_completion()
+	if(finished == REVSQUAD_VICTORY_REVS)
+		feedback_set_details("round_end_result","win - heads killed")
+		completion_text = "<br><span class='danger'><FONT size = 3> The heads of staff were killed or abandoned the station! The revolutionaries win!</FONT></span>"
+		stat_collection.revsquad.revsquad_won = 1
+	else if(finished == REVSQUAD_VICTORY_HEADS)
+		feedback_set_details("round_end_result","loss - rev heads killed")
+		completion_text = "<br><span class='danger'><FONT size = 3> The heads of staff managed to stop the revolution!</FONT></span>"
+	..()
+	return 1
+
+/datum/game_mode/proc/auto_declare_completion_revsquad()
+	var/list/targets = list()
+	var/text = ""
+	if(head_revolutionaries.len || istype(ticker.mode,/datum/game_mode/revolution))
+		var/icon/logo1 = icon('icons/mob/mob.dmi', "rev_head-logo")
+		end_icons += logo1
+		var/tempstate = end_icons.len
+		text += {"<img src="logo_[tempstate].png"> <FONT size = 2><B>The revolutionary squad members were:</B></FONT> <img src="logo_[tempstate].png">"}
+
+		for(var/datum/mind/headrev in head_revolutionaries)
+			if(headrev.current)
+				var/icon/flat = getFlatIcon(headrev.current, SOUTH, 1, 1)
+				end_icons += flat
+				tempstate = end_icons.len
+				text += {"<br><img src="logo_[tempstate].png"> <b>[headrev.key]</b> was <b>[headrev.name]</b> ("}
```

Comment:
It doesn't hurt anything, but I don't see why it is that way
Actually that applies to the overwhelming majority of these line notes but this one especially